### DISCUSSION TITLE
docker container 跨主機驗證註冊

### DISCUSSION THREAD

**Discussion by justin000abc** [dashingsoft/pyarmor#1642] [Q&A]
首先我在在已註冊的主機`192.168.5.5`
執行`pyarmor-auth pyarmor-device-regfile-6000.1.zip`

當我跨主機使用docker container進行驗證時，
`docker run --add-host=host.docker.internal:192.168.5.5 -it python bash`
在運行成功的container執行`pyarmor reg pyarmor-device-regfile-6000.1.zip`
terminal顯示註冊成功，
但當我開始進行加密`pyarmor gen -O /pyarmor /app`時，
termail卻顯示`ERROR invalid license token, try to run `pyarmor reg` to register license again`

**Comment by jondy** [comment]
请首先参阅文档
https://pyarmor.readthedocs.io/zh/latest/how-to/register.html#docker
并且升级到最新版本

**Reply by justin000abc** [reply]
我已經升級到最新版本
`pyarmor==8.4.1 pyarmor.cli.core==5.4.1`
並且執行過程都是按照文檔執行，
我不知道哪裡出錯

**Comment by jondy** [comment]
我也不知道你怎么认为 pyarmor 最新版本是 8.4.1
也不知道你怎么按照文档执行的，`Docker 主机和 Docker 容器必须在同一个网段，否则 Docker 主机无法接受到 Docker 容器的认证请求。`，这一点不知道你看到没有，确认 docker 容器的网址是否和 192.168.5.5 在同一个网段
不同主机类型可能需要不同的注册步骤，也不知道你看到了没有

**Reply by justin000abc** [reply]
我是linux主機
然後在你的文件上不就寫明

首先拷贝下面的文件到 Docker 主机:

pyarmor-8.4.1.tar.gz

pyarmor.cli.core-5.4.1-cp38-none-manylinux1_x86_64.whl

pyarmor.cli.core-5.4.1-cp311-none-manylinux1_x86_64.whl

pyarmor-device-regfile-6000.1.zip

那我認為我是照你的文件說的安裝8.4.1版本沒錯

**Reply by justin000abc** [reply]
然後我不知道你的網段是什麼意思，
是172.0.0.x，前三碼都要一樣嗎？

**Reply by justin000abc** [reply]
？？？
